system-image-upgrader

```diff
@@ -521,7 +521,11 @@ do
 
                 if [ -e partitions/${part}.img ] && [ -e $path ]; then
                     echo "Flashing ${part} at ${path}"
-                    cat partitions/${part}.img > ${path}
+                    if simgtest partitions/${part}.img; then
+                        simg2img partitions/${part}.img ${path}
```

Comment:
Have you tested that this flashes partition successfully? From quick look at simg2img code, it expects to write sparse files, which is a filesystem feature, not sure if this would work with block devices.

Reply:
Yes it works nicely. I stole the idea from another ROM, just lost the source now. But its actually smart enough it seems ;)